src/mlpack/methods/gmm/positive_definite_constraint.hpp

```diff
@@ -32,16 +32,23 @@ class PositiveDefiniteConstraint
    * and ensure each value on the diagonal is at least 1e-50.
    *
    * @param covariance Covariance matrix.
+   * @return bool True on success
    */
-  static void ApplyConstraint(arma::mat& covariance)
+  static bool ApplyConstraint(arma::mat& covariance)
```

Comment:
The return value of `ApplyConstraint()` is always ignored, though---so unless you want to refactor the `GMM` code to handle that (and I'm not sure how we would handle it), I would suggest here instead just sending a message via `Log::Warn` that the `PositiveDefiniteConstraint` could not be applied because the covariance matrix could not be eigendecomposed, or something like that.